runtime-modules/proposals/engine/Cargo.toml

```diff
@@ -16,13 +16,24 @@ sp-runtime = { package = 'sp-runtime', default-features = false, git = 'https://
 balances = { package = 'pallet-balances', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
 membership = { package = 'pallet-membership', default-features = false, path = '../../membership'}
 common = { package = 'pallet-common', default-features = false, path = '../../common'}
+frame-benchmarking = { package = 'frame-benchmarking', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca', optional = true}
+# We need to add the following dependencies(governance, recurringrrewards and minting) for benchmarking
```

Comment:
The only way I found to do that is to change [here](https://github.com/Joystream/joystream/pull/1731/files#diff-7f65dc0a272bf88b83e4d7361ba52d17af2e8ad908f0d9cab5c84b4ae70de091L41) by this:
```rs
#[cfg(features = "runtime-benchmarks")]
impl_outer_event! {
    pub enum TestEvent for Test {
        balances<T>,
        engine<T>,
        membership_mod<T>,
        frame_system<T>,
        council<T>,
    }
}

#[cfg(not(features = "runtime-benchmarks"))]
impl_outer_event! {
    pub enum TestEvent for Test {
        balances<T>,
        engine<T>,
        membership_mod<T>,
        frame_system<T>,
    }
}
```

Otherwise, it would break the test compilation when **not** using using `--features runtime-benchmarks`.

If you think that's the best way to go I will change it.

Reply:
Isn't it possible to put dependencies both in "dev-dependencies" and to the feature scope?

Reply:
No, because dependencies in "dev-dependencies" can't have `optional = true` and dependencies in the feature scope needs `optional = true`

Reply:
I was able to run successfully both `cargo test` and `cargo build --features runtime-benchmarks` with this config:

```
[package]
name = 'pallet-proposals-engine'
version = '4.0.1'
authors = ['Joystream contributors']
edition = '2018'

[dependencies]
serde = { version = "1.0.101", optional = true, features = ["derive"] }
codec = { package = 'parity-scale-codec', version = '1.3.4', default-features = false, features = ['derive'] }
sp-std = { package = 'sp-std', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
frame-support = { package = 'frame-support', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
frame-system = { package = 'frame-system', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
pallet-timestamp = { package = 'pallet-timestamp', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
sp-arithmetic = { package = 'sp-arithmetic', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
sp-runtime = { package = 'sp-runtime', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
balances = { package = 'pallet-balances', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
membership = { package = 'pallet-membership', default-features = false, path = '../../membership'}
common = { package = 'pallet-common', default-features = false, path = '../../common'}
frame-benchmarking = { package = 'frame-benchmarking', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca', optional = true}

# Benchmark dependencies.
governance = { package = 'pallet-governance', default-features = false, path = '../../governance', optional = true}
recurringrewards = { package = 'pallet-recurring-reward', default-features = false, path = '../../recurring-reward', optional = true}
minting = { package = 'pallet-token-mint', default-features = false, path = '../../token-minting', optional = true}

[dev-dependencies]
sp-io = { package = 'sp-io', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
sp-core = { package = 'sp-core', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
governance = { package = 'pallet-governance', default-features = false, path = '../../governance'}
recurringrewards = { package = 'pallet-recurring-reward', default-features = false, path = '../../recurring-reward'}
minting = { package = 'pallet-token-mint', default-features = false, path = '../../token-minting'}

[features]
default = ['std']
runtime-benchmarks = [
    'frame-benchmarking',
	'governance/std',
	'recurringrewards/std',
	'minting/std'
]
std = [
	'serde',
	'codec/std',
	'sp-std/std',
	'frame-support/std',
	'frame-system/std',
	'pallet-timestamp/std',
	'sp-arithmetic/std',
	'sp-runtime/std',
	'balances/std',
    'membership/std',
    'common/std',
]
```